File: 00. Estructura/01. Lenguajes de proposito general/04. Paradigmas de programación/02. Orientada a objetos/04. Sobrecarga/JavaScript/Sobrecarga.js

class Clase {
   
    // Se pueden clasificar metodos con el mismo nombre mientras  
    // no se repita la firma de los atributos (Mismo tipo, orden y tamaño)
    metodo(...args){
    	if (args.length == 0) {/*contendio*/} 																							// Firma 1 (Vacio)
        else if (args.length == 1 && args[0] instanceof tipo1) {/*contendio*/} 															// Firma 2 (tipo1)
        else if (args.length == 1 && args[0] instanceof tipo2) {/*contendio*/} 															// Firma 3 (tipo2)
        else if (args.length == 2 && args[0] instanceof tipo3 && args[1] instanceof tipo4) {/*contendio*/} 								// Firma 4 (tipo3, tipo4)
        else if (args.length == 2 && args[0] instanceof tipo5 && args[1] instanceof tipo6) {/*contendio*/} 								// Firma 5 (tipo5, tipo6)
        else if (args.length == N && args[0] instanceof tipo7 && args[1] instanceof tipo8 && args[N] instanceof tipo9) {/*contendio*/} 	// Firma 6 (tipo7, tipo8, tipo9)
        else if (args.length == N && args[0] instanceof tip10 && args[1] instanceof tip11 && args[N] instanceof tip12) {/*contendio*/} 	// Firma N (tip10, tip11, tip12)
    }
   
}

File: 00. Estructura/01. Lenguajes de proposito general/04. Paradigmas de programación/02. Orientada a objetos/04. Sobrecarga/Python/Sobrecarga.py

class Clase:
   
    # Se pueden clasificar metodos con el mismo nombre mientras  
    # no se repita la firma de los atributos (Mismo tipo, orden y tamaño)
    def metodo(*args):
    	if len(args) == 0: # Contenido																							          # Firma 1 (Vacio)
        elif len(args) == 1 and isinstance(args[0], tipo1): # Contenido    															      # Firma 2 (tipo1)
        elif len(args) == 1 and isinstance(args[0], tipo2): # Contenido    															      # Firma 3 (tipo2)
        elif len(args) == 2 and isinstance(args[0], tipo3) and isinstance(args[1], tipo4): # Contenido   							      # Firma 4 (tipo3, tipo4)
        elif len(args) == 2 and isinstance(args[0], tipo5) and isinstance(args[1], tipo6): # Contenido   								  # Firma 5 (tipo5, tipo6)
        elif len(args) == N and isinstance(args[0], tipo7) and isinstance(args[1], tipo8) and isinstance(args[N], tipo9): # Contenido  	  # Firma 6 (tipo7, tipo8, tipo9)
        elif len(args) == N and isinstance(args[0], tip10) and isinstance(args[1], tip11) and isinstance(args[N], tip12): # Contenido     # Firma N (tip10, tip11, tip12)
    
   

